### ISSUE TITLE
0.25 release not giving accurate responses/garbled. 

### ISSUE THREAD

**Issue by RussellSPG** [samtecspg/articulate#817]
### Issues
After going through the conversation interface and playing with different sayings/settings I'm concluding that with the implementation of the premade categories, that this release does not output accurate responses. Even after enabling training examples in settings, changing a response to something basic, it is not hitting the correct response. 

Here is an example

![image](https://user-images.githubusercontent.com/42249324/61229211-c68f7700-a6f5-11e9-9960-3992385a1219.png)

Though it may give the response it will combine small talk with order categories, and skips slots. 

It was never doing this or showed this much inaccuracy on the other releases, and I followed just as my script says until I was trying different ways to get better responses.

Source for response 1:
```
[
  {
    "docId": "J4IN9msBO0wEFMRBIjJ1",
    "category": "default",
    "project": "pizzabot break",
    "model": "default_63d7b018-fc9b-8403-3a0f-b7d2b3f1184d",
    "keywords": [
      {
        "start": 0,
        "end": 13,
        "value": {
          "value": "green peppers"
        },
        "confidence": 0.9282175311184978,
        "extractor": "ner_crf",
        "keyword": "toppings"
      },
      {
        "start": 18,
        "end": 30,
        "value": {
          "value": "black olives"
        },
        "confidence": 0.9571153609361837,
        "extractor": "ner_crf",
        "keyword": "toppings"
      }
    ],
    "action": {
      "name": "get order",
      "confidence": 0.9417749643325806
    },
    "action_ranking": [
      {
        "name": "get order",
        "confidence": 0.9417749643325806
      },
      {
        "name": "Marriage Proposal",
        "confidence": 0.1569221019744873
      },
      {
        "name": "By my Friend",
        "confidence": 0.07690344750881195
      },
      {
        "name": "Busy",
        "confidence": 0.07660019397735596
      },
      {
        "name": "Who Am I",
        "confidence": 0.06425304710865021
      },
      {
        "name": "You Suck",
        "confidence": 0.05246582254767418
      },
      {
        "name": "Capabilities",
        "confidence": 0.038409192115068436
      },
      {
        "name": "You're Welcome",
        "confidence": 0.023119542747735977
      },
      {
        "name": "order status",
        "confidence": 0
      },
      {
        "name": "Agree",
        "confidence": 0
      }
    ],
    "elapsed_time_ms": 45,
    "categoryScore": 0.964789628982544
  },
  {
    "docId": "J4IN9msBO0wEFMRBIjJ1",
    "category": "pizzabot break_modifiers",
    "project": "pizzabot break",
    "model": "pizzabot break_modifiers",
    "keywords": [],
    "action": {
      "name": "",
      "confidence": 0
    },
    "action_ranking": [],
    "elapsed_time_ms": 50,
    "categoryScore": 0
  }
]
```

Source after saying "I would like a pizza". It skips asking with a slot.

```
[
  {
    "docId": "K4JK9msBO0wEFMRB3TLB",
    "category": "default",
    "project": "pizzabot break",
    "model": "default_672158d4-78eb-5b93-15ec-94c287ebb0d8",
    "keywords": [],
    "action": {
      "name": "get order",
      "confidence": 0.9089829921722412
    },
    "action_ranking": [
      {
        "name": "get order",
        "confidence": 0.9089829921722412
      },
      {
        "name": "I like you",
        "confidence": 0.26332026720046997
      },
      {
        "name": "You Are Funny",
        "confidence": 0.17834246158599854
      },
      {
        "name": "order status",
        "confidence": 0.06005001813173294
      },
      {
        "name": "Busy",
        "confidence": 0.040516242384910583
      },
      {
        "name": "Sad",
        "confidence": 0.03349985182285309
      },
      {
        "name": "By my Friend",
        "confidence": 0.030392281711101532
      },
      {
        "name": "cancel order",
        "confidence": 0.02299197018146515
      },
      {
        "name": "chained-thanks",
        "confidence": 0.004334956407546997
      },
      {
        "name": "Who Am I",
        "confidence": 0.0005217790603637695
      }
    ],
    "elapsed_time_ms": 47,
    "categoryScore": 0.9371628761291504
  },
  {
    "docId": "K4JK9msBO0wEFMRB3TLB",
    "category": "pizzabot break_modifiers",
    "project": "pizzabot break",
    "model": "pizzabot break_modifiers",
    "keywords": [],
    "action": {
      "name": "",
      "confidence": 0
    },
    "action_ranking": [],
    "elapsed_time_ms": 40,
    "categoryScore": 0
  }
]
```

Terminal
```
]2019-07-15 14:38:42+0000 [-] "172.19.0.5" - - [15/Jul/2019:14:38:41 +0000] "POST /train?project=pizzabot+break&model=pizzabot+break_modifiers HTTP/1.1" 200 72 "-" "axios/0.18.0"
rasa_1           | 2019-07-15 14:38:42+0000 [-] "172.19.0.5" - - [15/Jul/2019:14:38:41 +0000] "DELETE /models?project=pizzabot+break&model=pizzabot+break_modifiers HTTP/1.1" 200 26 "-" "axios/0.18.0"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:14:38:42 +0000] "POST /api/agent/3/train HTTP/1.1" 200 815 "http://localhost:8080/agent/3/dialogue?filter=&page=17&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:39:38 +0000] "GET /api/agent/3/saying?skip=75&limit=5&field=id&direction=DESC&loadCategoryId=true HTTP/1.1" 200 547 "http://localhost:8080/agent/3/dialogue?filter=&page=17&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:39:38 +0000] "GET /api/agent/3/keyword?skip=0&limit=-1 HTTP/1.1" 200 891 "http://localhost:8080/agent/3/dialogue?filter=&page=17&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
rasa_1           | 2019-07-15 15:39:46+0000 [-] "172.19.0.5" - - [15/Jul/2019:15:39:46 +0000] "POST /parse HTTP/1.1" 200 755 "-" "axios/0.18.0"
rasa_1           | 2019-07-15 15:39:47+0000 [-] "172.19.0.5" - - [15/Jul/2019:15:39:46 +0000] "POST /parse HTTP/1.1" 200 1421 "-" "axios/0.18.0"
rasa_1           | 2019-07-15 15:39:47+0000 [-] "172.19.0.5" - - [15/Jul/2019:15:39:47 +0000] "POST /parse HTTP/1.1" 200 227 "-" "axios/0.18.0"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:39:47 +0000] "POST /api/agent/3/converse?debug=true HTTP/1.1" 200 948 "http://localhost:8080/agent/3/dialogue?filter=&page=17&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
rasa_1           | 2019-07-15 15:39:55+0000 [-] "172.19.0.5" - - [15/Jul/2019:15:39:55 +0000] "POST /parse HTTP/1.1" 200 389 "-" "axios/0.18.0"
rasa_1           | 2019-07-15 15:39:55+0000 [-] "172.19.0.5" - - [15/Jul/2019:15:39:55 +0000] "POST /parse HTTP/1.1" 200 1029 "-" "axios/0.18.0"
rasa_1           | 2019-07-15 15:39:55+0000 [-] "172.19.0.5" - - [15/Jul/2019:15:39:55 +0000] "POST /parse HTTP/1.1" 200 694 "-" "axios/0.18.0"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:39:56 +0000] "POST /api/agent/3/converse?debug=true HTTP/1.1" 200 928 "http://localhost:8080/agent/3/dialogue?filter=&page=17&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:09 +0000] "GET /api/agent/3/action?field=actionName&skip=0&limit=-1 HTTP/1.1" 200 1688 "http://localhost:8080/agent/3/action/51?filter=&page=16&pageSize=5&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:09 +0000] "GET /api/agent/3/keyword?skip=0&limit=-1 HTTP/1.1" 200 891 "http://localhost:8080/agent/3/action/51?filter=&page=16&pageSize=5&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:09 +0000] "GET /api/agent/3/action/51 HTTP/1.1" 200 469 "http://localhost:8080/agent/3/action/51?filter=&page=16&pageSize=5&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:09 +0000] "GET /api/agent/3/keyword?skip=0&limit=-1 HTTP/1.1" 200 891 "http://localhost:8080/agent/3/action/51?filter=&page=16&pageSize=5&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:09 +0000] "GET /api/agent/3/action?field=actionName&skip=0&limit=-1 HTTP/1.1" 200 1688 "http://localhost:8080/agent/3/action/51?filter=&page=16&pageSize=5&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:09 +0000] "GET /api/agent/3/action?field=actionName&skip=0&limit=-1 HTTP/1.1" 200 1688 "http://localhost:8080/agent/3/action/51?filter=&page=16&pageSize=5&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:18 +0000] "PUT /api/agent/3/action/51 HTTP/1.1" 200 469 "http://localhost:8080/agent/3/action/51?filter=&page=16&pageSize=5&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:20 +0000] "GET /api/agent/3/category?skip=0&limit=-1 HTTP/1.1" 200 530 "http://localhost:8080/agent/3/dialogue?filter=&page=16&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:20 +0000] "GET /api/agent/3/action?field=actionName&skip=0&limit=-1 HTTP/1.1" 200 1688 "http://localhost:8080/agent/3/dialogue?filter=&page=16&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:20 +0000] "GET /api/agent/3/action?field=actionName&skip=0&limit=-1 HTTP/1.1" 200 1688 "http://localhost:8080/agent/3/dialogue?filter=&page=16&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:20 +0000] "GET /api/agent/3/action?field=actionName&skip=0&limit=-1 HTTP/1.1" 200 1688 "http://localhost:8080/agent/3/dialogue?filter=&page=16&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:20 +0000] "GET /api/agent/3/saying?skip=75&limit=5&field=id&direction=DESC&loadCategoryId=true HTTP/1.1" 200 547 "http://localhost:8080/agent/3/dialogue?filter=&page=16&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:21 +0000] "GET /api/agent/3/keyword?skip=0&limit=-1 HTTP/1.1" 200 891 "http://localhost:8080/agent/3/dialogue?filter=&page=16&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
rasa_1           | 2019-07-15 15:40:25+0000 [-] "172.19.0.5" - - [15/Jul/2019:15:40:24 +0000] "POST /parse HTTP/1.1" 200 377 "-" "axios/0.18.0"
rasa_1           | 2019-07-15 15:40:25+0000 [-] "172.19.0.5" - - [15/Jul/2019:15:40:24 +0000] "POST /parse HTTP/1.1" 200 189 "-" "axios/0.18.0"
rasa_1           | 2019-07-15 15:40:25+0000 [-] "172.19.0.5" - - [15/Jul/2019:15:40:24 +0000] "POST /parse HTTP/1.1" 200 1029 "-" "axios/0.18.0"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:25 +0000] "POST /api/agent/3/converse?debug=true HTTP/1.1" 200 955 "http://localhost:8080/agent/3/dialogue?filter=&page=16&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"
rasa_1           | 2019-07-15 15:40:29+0000 [-] "172.19.0.5" - - [15/Jul/2019:15:40:29 +0000] "POST /parse HTTP/1.1" 200 395 "-" "axios/0.18.0"
rasa_1           | 2019-07-15 15:40:30+0000 [-] "172.19.0.5" - - [15/Jul/2019:15:40:29 +0000] "POST /parse HTTP/1.1" 200 207 "-" "axios/0.18.0"
rasa_1           | 2019-07-15 15:40:30+0000 [-] "172.19.0.5" - - [15/Jul/2019:15:40:29 +0000] "POST /parse HTTP/1.1" 200 1062 "-" "axios/0.18.0"
nginx_1          | 172.19.0.1 - - [15/Jul/2019:15:40:30 +0000] "POST /api/agent/3/converse?debug=true HTTP/1.1" 200 989 "http://localhost:8080/agent/3/dialogue?filter=&page=16&tab=sayings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0" "-"

```

No console errors.

- Did you download a release or pull from Master? Which release?
release 0.25
- What operating system are you running?
10 Pro
- Are you running locally or in the cloud?
local
- Which browser are you using?
FF
----

**Comment by wrathagom** [comment]
Can you tell us where in the picture it goes wrong? I can't tell what isn't correct. What your script says should happen vs what is happening?

**Comment by RussellSPG** [comment]
So when I Said, green peppers and black olives, it wouldn't say "I have a green peppers, black olive pizza" it would just say the fallback. I had to redo the training several times, and change the response. I checked handlebars etc so it wasn't the response being wrong it just would shoot out inaccurate responses. It seems it was always throwing out "small talk" responses more than anything whenever I asked something. This behavior wasn't happening on any other releases really..
